enhancements/cluster-logging/forwarding-json-structured-logs.md

```diff
@@ -84,155 +85,138 @@ Logging terminology can have overlapping meanings, this document uses the follow
 
 ### Non-Goals
 
 * General-purpose JSON queries and transformations.
-* Recording or validating user-defined schema.
-  The forwarder only identifies schema by name, all schema knowledge belongs to the user.
 
 ## Proposal
 
 ### Data model
 
-Two new top-level fields added to the logging output record [data model][data_model]:
+One new top-level fields added to the logging output record [data model][data_model]:
 
 * `structured` (type object): Original structured JSON log entry.
-* `schema` (type string): Schema name assigned by user.
 
 Relationship between `structured` and `message` fields:
 
-* EXACTLY ONE of `message` or `structured` MUST be present.
-* The presence of `structured` or `message` fields identifies structured/unstructured records.
-* The `schema` field is optional.
-  It is provided to represent the schema of a structured record, but its use is up to the consumer.
-
-For the default *Elasticsearch* store, `schema` is used to generate the index name.
+* `message` field is always present for backwards compatibility (a future proposal will address filtering out unwanted fields in the record)
+* Both `message` and `structured` fields may be present.
+* If there is no structured data, the `structured` field will be missing or empty
+* If `message` and `structured` are both present and non-empty, `message` MUST contain the JSON-quoted string equivalent of the `structured` value.
 
-For the initial implementation the forwarder will use the `schema` name to generate the `viaq_index_name` field.
-Future versions of the Elasticsearch store may use `schema` directly.
+*Elasticsearch* outputs MAY specify special index names for records, see below.
 
 #### ClusterLogForwarder configuration
 
-New *pipeline* fields:
+New *pipeline* field:
 
-* `parse`: (string, optional) Legal values are "JSON" or "json" (there may be others in future).
+* `parse`: (string, optional) Only legal value is "json" (there may be others in future).
    If set, attempt to parse log entries as JSON objects.
-* `schemaKey`: (string, optional) Use value of meta-data key as `schema` value, if present.
+
+For each log entry; if `parse: json` is set _and_ the entry is valid JSON, the output record will include a `structured` field _equivalent_ to the JSON entry.
+It may differ in field order and use of white-space.
+
+All record will have a `message` field for backwards compatibility.
+
+#### Output  configuration for type:elasticsearch
+
+New fields for output type `elasticsearch`
+* `indexKey`: (string, optional) Use value of meta-data key as `index` value, if present.
   These keys are supported:
-  - `kubernetes.namespaceName`: Use the namespace name as the schema name.
+  - `kubernetes.namespaceName`: Use the namespace name as the index name.
   - `kubernetes.labels.<key>`: Use the string value of kubernetes label with key `<key>`.
+  - `openshift.labels.<key>`: Use the string value of an openshift label with key `<key>` (see [](forwarder-tagging.md))
```

Comment:
What is this?

Reply:
Its a markdown link to the forwarder-tagging.md proposal. If you render the markdown as HTML (e.g. view it on github) it becomes a real link.